app/assets/javascripts/analytics/ecommerce.js

```diff
@@ -23,18 +24,22 @@
         var contentId = $ecommerceRow.attr('data-ecommerce-content-id'),
           path = $ecommerceRow.attr('data-ecommerce-path');
 
-        addImpression(contentId, path, index + startPosition, searchQuery, listTitle, variant);
-        trackProductOnClick($ecommerceRow, contentId, path, index + startPosition, searchQuery, listTitle, variant, trackClickLabel);
+        addImpression(contentId, path, index + startPosition, searchQuery, listTitle, listSubheading, variant);
+        trackProductOnClick($ecommerceRow, contentId, path, index + startPosition, searchQuery, listTitle, listSubheading, variant, trackClickLabel);
       });
     }
 
-    function constructData(contentId, path, position, listTitle, searchQuery, variant) {
+    function constructData(contentId, path, position, listTitle, listSubheading, searchQuery, variant) {
       var data = {
         position: position,
         list: listTitle,
         dimension71: searchQuery
       }
 
+      if (listSubheading !== undefned) {
```

Comment:
I think you can also just do `if (listSubheading)`, which will catch the null case too? Hmm, but will it ever be an empty string?

But otherwise, there's a typo:
```suggestion
      if (listSubheading !== undefined) {
```
Is this code tested anywhere?

Reply:
Fixed, thanks!